src/ol/layers/Layer.js

```diff
@@ -67,16 +67,20 @@ class Layer extends LayerCommon {
     if (visible === this.visible) {
       return;
     }
+
+    this.visible = visible;
+
     if (this.olLayer) {
       this.olLayer.setVisible(this.visible);
     }
 
-    super.setVisible(
-      visible,
+    this.dispatchEvent({
```

Comment:
does it work ?

Reply:
It did so because it was broken when integrated in react-spatial.

'this.visible = visible'
It was set in the common Layer, but it must be called before
'this.olLayer.setVisible(this.visible);'
Calling with super created a timing pb, and splitting the setVisible is easier.

Reply:
what if you call the super.setVisible before  
```
if (this.olLayer) {
￼      this.olLayer.setVisible(this.visible)
}
```

Does it work ? it should

Reply:
Indeed I did as so :)